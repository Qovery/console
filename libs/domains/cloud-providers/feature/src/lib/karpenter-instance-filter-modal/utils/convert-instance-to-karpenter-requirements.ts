import {
  type ClusterInstanceTypeResponseListResultsInner,
  type KarpenterNodePoolRequirement,
} from 'qovery-typescript-axios'

export function convertToKarpenterRequirements(
  instance: ClusterInstanceTypeResponseListResultsInner[]
): KarpenterNodePoolRequirement[] {
  if (!instance || instance.length === 0) {
    return []
  }

  const requirements: KarpenterNodePoolRequirement[] = []

  const uniqueValues = {
    architectures: new Set<string>(),
    families: new Set<string>(),
    sizes: new Set<string>(),
  }

  instance.forEach((instance) => {
    if (instance.architecture) {
      uniqueValues.architectures.add(instance.architecture)
    }

    if (instance.attributes?.instance_family) {
      uniqueValues.families.add(instance.attributes.instance_family)
    }
    if (instance.attributes?.instance_size) {
      uniqueValues.sizes.add(instance.attributes.instance_size)
    }
  })

  if (uniqueValues.sizes.size > 0) {
    requirements.push({
      key: 'InstanceSize',
      operator: 'In',
      values: Array.from(uniqueValues.sizes),
    })
  }

  if (uniqueValues.architectures.size > 0) {
    requirements.push({
      key: 'Arch',
      operator: 'In',
      values: Array.from(uniqueValues.architectures),
    })
  }

  if (uniqueValues.families.size > 0) {
    requirements.push({
      key: 'InstanceFamily',
      operator: 'In',
      values: Array.from(uniqueValues.families),
    })
  }

  return requirements
}
