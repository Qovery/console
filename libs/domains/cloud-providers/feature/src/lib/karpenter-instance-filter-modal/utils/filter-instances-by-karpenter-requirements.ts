import { type ClusterInstanceTypeResponseListResultsInner } from 'qovery-typescript-axios'
import { type KarpenterData } from '@qovery/shared/interfaces'

export function filterInstancesByKarpenterRequirements(
  allInstances: ClusterInstanceTypeResponseListResultsInner[],
  karpenterData: Omit<KarpenterData, 'disk_size_in_gib' | 'enabled' | 'spot_enabled'>
): ClusterInstanceTypeResponseListResultsInner[] {
  const karpenterRequirements = karpenterData.qovery_node_pools?.requirements || []

  return allInstances.filter((instance) => {
    const instanceSize = instance.attributes?.instance_size || ''
    const InstanceFamily = instance.attributes?.instance_family || ''
    const instanceArchitecture = instance.architecture || ''

    return karpenterRequirements.every((requirement) => {
      if (requirement.key === 'InstanceSize') {
        return requirement.values.includes(instanceSize)
      }

      if (requirement.key === 'Arch') {
        // Fix inconsistency between uppercase and lowercase with architecture during the migration
        return requirement.values.map((v) => v.toUpperCase()).includes(instanceArchitecture)
      }

      if (requirement.key === 'InstanceFamily') {
        return requirement.values.includes(InstanceFamily)
      }

      return false
    })
  })
}

export default filterInstancesByKarpenterRequirements
