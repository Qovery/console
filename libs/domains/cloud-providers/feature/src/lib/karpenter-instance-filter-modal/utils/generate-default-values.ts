import { type ClusterInstanceTypeResponseListResultsInner } from 'qovery-typescript-axios'
import { type KarpenterInstanceFormProps } from '../karpenter-instance-filter-modal'

export function generateDefaultValues(
  instances: ClusterInstanceTypeResponseListResultsInner[]
): Omit<KarpenterInstanceFormProps, 'default_service_architecture'> {
  // Get unique architectures
  const architectures = [...new Set(instances.map((i) => i.architecture))]

  // Get unique sizes without sorting
  const sizes = [...new Set(instances.map((i) => i.attributes?.instance_size ?? ''))].filter((i) => i !== '')

  // Group instance families by raw instance_category
  const categories: Record<string, string[]> = {}
  instances.forEach((instance) => {
    const family = instance.attributes?.instance_family || ''
    const category = instance.attributes?.instance_category || ''

    if (!categories[category]) {
      categories[category] = []
    }

    if (!categories[category].includes(family)) {
      categories[category].push(family)
    }
  })

  return {
    ARM64: architectures.includes('ARM64'),
    AMD64: architectures.includes('AMD64'),
    categories,
    sizes,
  }
}

export default generateDefaultValues
