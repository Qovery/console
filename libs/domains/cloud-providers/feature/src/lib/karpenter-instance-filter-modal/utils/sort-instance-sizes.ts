type BasicSize = 'small' | 'medium' | 'large' | 'xlarge'
type SizePriority = Record<BasicSize, number>

// Define type for the sort key tuple
type SortKey = [priority: number, multiplier: number]

export function sortInstanceSizes(sizes: string[]): string[] {
  const sizePriority: SizePriority = {
    small: 1,
    medium: 2,
    large: 3,
    xlarge: 4,
  }

  function getSortKey(size: string): SortKey {
    if (isBasicSize(size)) {
      return [sizePriority[size], 0]
    }

    // Handle xlarge variants
    if (size.includes('xlarge')) {
      // Extract number before 'xlarge'
      const number = size.replace('xlarge', '')
      // Convert empty string to 1 (for just 'xlarge')
      const multiplier = number ? parseInt(number) : 1
      return [5, multiplier]
    }

    return [0, 0]
  }

  function isBasicSize(size: string): size is BasicSize {
    return size in sizePriority
  }

  return sizes.sort((a, b) => {
    const [priorityA, multiplierA] = getSortKey(a)
    const [priorityB, multiplierB] = getSortKey(b)

    if (priorityA !== priorityB) {
      return priorityA - priorityB
    }
    return multiplierA - multiplierB
  })
}

export default sortInstanceSizes
