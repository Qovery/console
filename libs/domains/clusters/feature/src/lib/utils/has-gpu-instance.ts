import { type Cluster } from 'qovery-typescript-axios'

export const hasGpuInstance = (cluster?: Cluster) => {
  const clusterFeatureKarpenter = cluster?.features?.find((feature) => feature.id === 'KARPENTER')
  if (!clusterFeatureKarpenter) return false
  return Boolean(
    typeof clusterFeatureKarpenter?.value_object?.value === 'object' &&
      'qovery_node_pools' in clusterFeatureKarpenter.value_object.value &&
      clusterFeatureKarpenter.value_object.value.qovery_node_pools.gpu_override
  )
}
