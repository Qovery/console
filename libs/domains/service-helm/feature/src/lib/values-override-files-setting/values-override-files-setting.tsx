import { type GitProviderEnum, type GitTokenResponse, type HelmRequestAllOfSource } from 'qovery-typescript-axios'
import { type PropsWithChildren, type ReactNode } from 'react'
import { Controller, type UseFormReturn } from 'react-hook-form'
import { SettingsHeading } from '@qovery/shared/console-shared'
import { ExternalLink, Heading, Icon, InputSelect, Popover, Section } from '@qovery/shared/ui'
import ValuesOverrideYamlSetting from '../values-override-yaml-setting/values-override-yaml-setting'

export type ValuesOverrideTypes = 'GIT_REPOSITORY' | 'YAML' | 'NONE'

export interface HelmValuesFileData {
  type: ValuesOverrideTypes
  repository?: string | null
  provider?: keyof typeof GitProviderEnum
  git_token_id?: GitTokenResponse['id']
  git_token_name?: GitTokenResponse['name']
  is_public_repository?: boolean
  branch?: string
  paths?: string
  content?: string
  auto_deploy?: boolean
}

export interface ValuesOverrideFilesSettingProps {
  source: HelmRequestAllOfSource
  methods: UseFormReturn<HelmValuesFileData>
  watchFieldType: ValuesOverrideTypes
  gitRepositorySettings: ReactNode
  onSubmit: () => void
  isSetting?: boolean
}

export function ValuesOverrideFilesSetting({
  methods,
  source,
  watchFieldType,
  children,
  gitRepositorySettings,
  onSubmit,
  isSetting,
}: PropsWithChildren<ValuesOverrideFilesSettingProps>) {
  return (
    <Section>
      {isSetting ? (
        <SettingsHeading
          title="Values override as file"
          description="Define the YAML file(s) to be applied as override to the default values.yaml delivered with the chart. It is highly recommended to store the override file(s) in a git repository."
        />
      ) : (
        <Heading className="mb-2">Values override as file</Heading>
      )}

      <form className="space-y-10" onSubmit={onSubmit}>
        {!isSetting && (
          <div>
            <p className="text-sm text-neutral-350">
              Define the YAML file(s) to be applied as override to the default values.yaml delivered with the chart. It
              is highly recommended to store the override file(s) in a git repository.
            </p>
            <Popover.Root>
              <Popover.Trigger>
                <span className="cursor-pointer text-sm font-medium text-brand-500 transition hover:text-brand-600">
                  How it works <Icon className="text-xs" iconStyle="regular" iconName="circle-question" />
                </span>
              </Popover.Trigger>
              <Popover.Content side="left" className="relative text-sm text-neutral-350" style={{ width: 440 }}>
                <h6 className="mb-2 font-medium text-neutral-400">How it works</h6>
                <ul className="list-disc pl-4">
                  <li>
                    Your helm chart might have already a variables.yaml file with some basic configuration. In this
                    section you can define your own overrides to customize the helm chart behaviour.
                  </li>
                  <li>
                    You can define the overrides by selecting a YAML file from a git repository (preferred) or by
                    passing a raw YAML file.
                  </li>
                  <li>
                    If you don’t have a file, you can skip this step and instead define the values override directly as
                    arguments (--set).
                  </li>
                  <li>
                    You can use the Qovery environment variables as overrides by using the placeholder
                    “qovery.env.ENV_VAR_NAME” (Example: qovery.env.DB_URL. Qovery will manage the replacement of those
                    placeholders at deployment time.
                  </li>
                  <li>
                    To let you access every Qovery functionality, additional Qovery labels and annotations are
                    automatically injected in some of the Kubernetes objects deployed within your helm.
                  </li>
                </ul>
                <ExternalLink href="https://hub.qovery.com/docs/using-qovery/configuration/helm/#values">
                  Documentation
                </ExternalLink>
                <Popover.Close className="absolute right-4 top-4">
                  <button type="button">
                    <Icon iconName="xmark" className="text-lg leading-4 text-neutral-400" />
                  </button>
                </Popover.Close>
              </Popover.Content>
            </Popover.Root>
          </div>
        )}
        <div className="flex flex-col gap-4">
          <Controller
            name="type"
            control={methods.control}
            defaultValue="GIT_REPOSITORY"
            render={({ field }) => (
              <InputSelect
                label="File source"
                value={field.value}
                onChange={field.onChange}
                options={[
                  {
                    label: 'Git repository',
                    value: 'GIT_REPOSITORY',
                  },
                  {
                    label: 'Raw YAML',
                    value: 'YAML',
                  },
                  {
                    label: 'None',
                    value: 'NONE',
                  },
                ]}
              />
            )}
          />
        </div>
        {watchFieldType === 'YAML' && (
          <Section>
            <Heading className="mb-2">Override with raw Yaml</Heading>
            <p className="mb-6 text-sm text-neutral-350">
              You can define here the YAML containing the overrides you want to apply. The YAML will be stored by Qovery
              and can be updated later within the settings but no history will be retained.
            </p>
            <div className="flex flex-col gap-3">
              <ValuesOverrideYamlSetting
                content={methods.getValues('content')}
                onSubmit={(value) => {
                  methods.setValue('content', value)
                  isSetting && onSubmit()
                }}
                source={source}
              />
            </div>
          </Section>
        )}

        {watchFieldType === 'GIT_REPOSITORY' && (
          <Section>
            <Heading className="mb-2">Override from repository</Heading>
            <p className="mb-6 text-sm text-neutral-350">
              Specify the repository and the path containing the override yaml file to be passed via the “-f” helm
              argument. More than one file can be used as override by adding them in the path field separated by a
              semi-colon. If you don’t have a repository, you can set the override manually or via a raw YAML file.
            </p>
            <div className="flex flex-col gap-3">{gitRepositorySettings}</div>
          </Section>
        )}
        {children}
      </form>
    </Section>
  )
}

export default ValuesOverrideFilesSetting
