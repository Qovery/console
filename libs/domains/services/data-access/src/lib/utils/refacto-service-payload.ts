import {
  type ApplicationEditRequest,
  type ApplicationGitRepositoryRequest,
  DatabaseModeEnum,
  type HelmRequest,
  type HelmResponseAllOfSourceOneOf,
  type HelmResponseAllOfSourceOneOf1,
  type ServiceStorageStorageInner,
} from 'qovery-typescript-axios'
import { match } from 'ts-pattern'
import { isJobGitSource } from '@qovery/shared/enums'
import { type Application, type Container, type Database, type Helm, type Job } from '../domains-services-data-access'

/* 
TODO: all this following functions should be removed after the API refactoring and we need to clean it after the Services migration to React Query 
https://www.notion.so/qovery/API-improvements-b54ba305c2ee4e549eb002278c532c7f?pvs=4#7d71ea23b5fa44ca80c15a0c32ebd8da
*/

export function refactoApplication(application: Application) {
  // refacto because we can't send all git data
  if (application.git_repository) {
    application.git_repository = {
      url: application.git_repository.url,
      branch: application.git_repository.branch,
      root_path: application.git_repository.root_path,
      git_token_id: application.git_repository.git_token_id,
    }
  }

  // refacto to remove the id by storage
  if (application.storage) {
    application.storage =
      application.storage.length > 0
        ? application.storage.map((storage: ServiceStorageStorageInner) => ({
            mount_point: storage.mount_point,
            size: storage.size,
            type: storage.type,
            id: storage.id,
          }))
        : []
  }

  const applicationRequestPayload: ApplicationEditRequest = {
    name: application.name,
    storage: application.storage,
    cpu: application.cpu,
    git_repository: application.git_repository as ApplicationGitRepositoryRequest,
    build_mode: application.build_mode,
    description: application.description || undefined,
    memory: application.memory,
    auto_preview: application.auto_preview,
    auto_deploy: application.auto_deploy,
    ports: application.ports,
    dockerfile_path: application.dockerfile_path || undefined,
    healthchecks: application.healthchecks ?? {},
    buildpack_language: application.buildpack_language,
    max_running_instances: application.max_running_instances,
    min_running_instances: application.min_running_instances,
    entrypoint: application.entrypoint,
    arguments: application.arguments,
  }

  return applicationRequestPayload
}

export function refactoContainer(container: Container) {
  const containerRequestPayload = {
    name: container.name || '',
    description: container.description || '',
    storage: container.storage,
    ports: container.ports,
    cpu: container.cpu,
    memory: container.memory,
    max_running_instances: container.max_running_instances,
    min_running_instances: container.min_running_instances,
    registry_id: container.registry?.id || '',
    image_name: container.image_name || '',
    tag: container.tag || '',
    arguments: container.arguments,
    entrypoint: container.entrypoint,
    auto_preview: container.auto_preview,
    auto_deploy: container.auto_deploy,
    healthchecks: container.healthchecks,
  }

  return containerRequestPayload
}

export function refactoJob(job: Job) {
  const jobRequest = {
    name: job.name,
    description: job.description,
    cpu: job.cpu,
    memory: job.memory,
    auto_preview: job.auto_preview,
    auto_deploy: job.auto_deploy,
    max_duration_seconds: job.max_duration_seconds,
    port: job.port,
    max_nb_restart: job.max_nb_restart,
    schedule: job.schedule,
    healthchecks: job.healthchecks ?? {},
    source: {},
  }

  if (isJobGitSource(job.source)) {
    jobRequest.source = {
      docker: {
        dockerfile_path: job.source.docker?.dockerfile_path,
        git_repository: {
          url: job.source.docker?.git_repository?.url || '',
          branch: job.source.docker?.git_repository?.branch,
          root_path: job.source.docker?.git_repository?.root_path,
          git_token_id: job.source.docker?.git_repository?.git_token_id,
        },
      },
    }
  } else {
    jobRequest.source = {
      image: {
        registry_id: job.source?.image?.registry_id,
        image_name: job.source?.image?.image_name,
        tag: job.source?.image?.tag,
      },
    }
  }

  return jobRequest
}

export function refactoDatabase(database: Database) {
  const databaseRequestPayload = {
    name: database.name,
    description: database.description,
    version: database.version,
    accessibility: database.accessibility,
    cpu: database.cpu,
    memory: database.memory,
    storage: database.storage,
    instance_type: database.instance_type,
  }

  if (database.mode === DatabaseModeEnum.MANAGED) databaseRequestPayload.version = database.version

  return databaseRequestPayload
}

export function refactoHelm(helm: Helm) {
  const sourceProvider = (helm?.source as HelmResponseAllOfSourceOneOf).git ? 'GIT' : 'HELM_REPOSITORY'

  const source = match(sourceProvider)
    .with('GIT', () => {
      const gitRepository = (helm?.source as HelmResponseAllOfSourceOneOf).git?.git_repository
      return {
        git_repository: {
          url: gitRepository?.url ?? '',
          branch: gitRepository?.branch,
          root_path: gitRepository?.root_path,
          git_token_id: gitRepository?.git_token_id,
        },
      }
    })
    .with('HELM_REPOSITORY', () => {
      const helmRepository = (helm?.source as HelmResponseAllOfSourceOneOf1).repository
      return {
        helm_repository: {
          repository: helmRepository?.repository?.id,
          chart_name: helmRepository?.chart_name,
          chart_version: helmRepository?.chart_version,
        },
      }
    })
    .exhaustive()

  const helmRequestPayload: HelmRequest = {
    name: helm.name,
    description: helm.description,
    auto_preview: helm.auto_preview,
    auto_deploy: helm.auto_deploy,
    ports: helm.ports,
    allow_cluster_wide_resources: helm.allow_cluster_wide_resources,
    arguments: helm.arguments,
    values_override: helm.values_override,
    source,
  }

  return helmRequestPayload
}
