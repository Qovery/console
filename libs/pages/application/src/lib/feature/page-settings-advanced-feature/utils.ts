import { ApplicationAdvancedSettings } from 'qovery-typescript-axios'
import { ApplicationEntity } from '@console/shared/interfaces'

export function initFormValues(keys: string[], application: ApplicationEntity): { [key: string]: string } {
  const values: { [key: string]: string } = {}
  keys.forEach((key) => {
    if (application.advanced_settings?.current_settings) {
      values[key] =
        application.advanced_settings.current_settings[key as keyof ApplicationAdvancedSettings]?.toString() || ''
    }
  })

  return values
}
