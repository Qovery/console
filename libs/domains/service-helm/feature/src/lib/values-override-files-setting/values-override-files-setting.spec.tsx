import { wrapWithReactHookForm } from '__tests__/utils/wrap-with-react-hook-form'
import { renderWithProviders } from '@qovery/shared/util-tests'
import { ValuesOverrideFilesSetting, type ValuesOverrideFilesSettingProps } from './values-override-files-setting'

describe('ValuesOverrideFilesSetting', () => {
  const props: ValuesOverrideFilesSettingProps = {
    type: 'NONE',
    source: {
      git_repository: {
        url: 'github.com',
        branch: 'main',
        root_path: 'root',
        git_token_id: 'token',
      },
    },
    methods: {
      handleSubmit: jest.fn(),
      watch: jest.fn(),
      getValues: jest.fn(),
      setValue: jest.fn(),
      register: jest.fn(),
      unregister: jest.fn(),
      formState: {
        errors: {},
        isDirty: false,
        isValid: false,
        isSubmitting: false,
        touched: {},
        submitCount: 0,
      },
    },
    watchFieldType: 'NONE',
    gitRepositorySettings: <div>git repository dom</div>,
  }

  it('should match snapshot with NONE type', () => {
    const { baseElement } = renderWithProviders(wrapWithReactHookForm(<ValuesOverrideFilesSetting {...props} />))
    expect(baseElement).toMatchSnapshot()
  })

  it('should match snapshot with YAML type', () => {
    const { baseElement } = renderWithProviders(
      wrapWithReactHookForm(<ValuesOverrideFilesSetting {...props} watchFieldType="YAML" />)
    )
    expect(baseElement).toMatchSnapshot()
  })

  it('should match snapshot with GIT_REPOSITORY type', () => {
    const { baseElement } = renderWithProviders(
      wrapWithReactHookForm(<ValuesOverrideFilesSetting {...props} watchFieldType="GIT_REPOSITORY" />)
    )
    expect(baseElement).toMatchSnapshot()
  })
})
