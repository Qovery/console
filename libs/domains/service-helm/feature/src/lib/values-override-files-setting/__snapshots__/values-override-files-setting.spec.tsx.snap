// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ValuesOverrideFilesSetting should match snapshot with GIT_REPOSITORY type 1`] = `
<body>
  <div>
    <section
      class="flex flex-col"
    >
      <h1
        class="text-neutral-400 text-xl font-bold mb-2"
      >
        Values override as file
      </h1>
      <form
        class="space-y-10"
      >
        <div>
          <p
            class="text-sm text-neutral-350"
          >
            Define the YAML file(s) to be applied as override to the default values.yaml delivered with the chart. It is highly recommended to store the override file(s) in a git repository.
          </p>
          <span
            aria-controls="radix-:rk:"
            aria-expanded="false"
            aria-haspopup="dialog"
            class="cursor-pointer text-sm font-medium text-brand-500 transition hover:text-brand-600"
            data-state="closed"
            type="button"
          >
            How it works 
            <i
              aria-hidden="true"
              class="fa-regular fa-circle-question text-xs"
            />
          </span>
        </div>
        <div
          class="flex flex-col gap-4"
        >
          <div
            class=""
          >
            <div
              class="input input--select    "
              data-testid="select"
            >
              <label
                class="!translate-y-0 !text-xs"
                for="File source"
              >
                File source
              </label>
              <div
                class=" css-b62m3t-container"
              >
                <span
                  class="css-1f43avz-a11yText-A11yText"
                  id="react-select-4-live-region"
                />
                <span
                  aria-atomic="false"
                  aria-live="polite"
                  aria-relevant="additions text"
                  class="css-1f43avz-a11yText-A11yText"
                />
                <div
                  class="input-select__control css-13cymwt-control"
                >
                  <div
                    class="input-select__value-container input-select__value-container--has-value css-1fdsijx-ValueContainer"
                  >
                    <span
                      class="mr-1 text-sm text-neutral-400"
                    >
                      Git repository
                    </span>
                    <input
                      aria-autocomplete="list"
                      aria-expanded="false"
                      aria-haspopup="true"
                      aria-readonly="true"
                      class="css-mohuvp-dummyInput-DummyInput"
                      id="File source"
                      inputmode="none"
                      role="combobox"
                      tabindex="0"
                      value=""
                    />
                  </div>
                  <div
                    class="input-select__indicators css-1hb7zxy-IndicatorsContainer"
                  >
                    <span
                      class="input-select__indicator-separator css-1u9des2-indicatorSeparator"
                    />
                    <div
                      aria-hidden="true"
                      class="input-select__indicator input-select__dropdown-indicator css-1xc3v61-indicatorContainer"
                    >
                      <svg
                        aria-hidden="true"
                        class="css-tj5bde-Svg"
                        focusable="false"
                        height="20"
                        viewBox="0 0 20 20"
                        width="20"
                      >
                        <path
                          d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
                <input
                  name="File source"
                  type="hidden"
                  value="GIT_REPOSITORY"
                />
              </div>
              <input
                name="File source"
                type="hidden"
                value="GIT_REPOSITORY"
              />
              <div
                class="pointer-events-none absolute right-4 top-1/2 -translate-y-1/2"
              >
                <span
                  class="icon-solid-angle-down shrink-0 text-sm text-neutral-400"
                  role="img"
                />
              </div>
            </div>
          </div>
        </div>
        <section
          class="flex flex-col"
        >
          <h2
            class="text-neutral-400 text-base font-bold mb-2"
          >
            Override from repository
          </h2>
          <p
            class="mb-6 text-sm text-neutral-350"
          >
            Specify the repository and the path containing the override yaml file to be passed via the “-f” helm argument. More than one file can be used as override by adding them in the path field separated by a semi-colon. If you don’t have a repository, you can set the override manually or via a raw YAML file.
          </p>
          <div
            class="flex flex-col gap-3"
          >
            <div>
              git repository dom
            </div>
          </div>
        </section>
      </form>
    </section>
  </div>
</body>
`;

exports[`ValuesOverrideFilesSetting should match snapshot with NONE type 1`] = `
<body>
  <div>
    <section
      class="flex flex-col"
    >
      <h1
        class="text-neutral-400 text-xl font-bold mb-2"
      >
        Values override as file
      </h1>
      <form
        class="space-y-10"
      >
        <div>
          <p
            class="text-sm text-neutral-350"
          >
            Define the YAML file(s) to be applied as override to the default values.yaml delivered with the chart. It is highly recommended to store the override file(s) in a git repository.
          </p>
          <span
            aria-controls="radix-:r6:"
            aria-expanded="false"
            aria-haspopup="dialog"
            class="cursor-pointer text-sm font-medium text-brand-500 transition hover:text-brand-600"
            data-state="closed"
            type="button"
          >
            How it works 
            <i
              aria-hidden="true"
              class="fa-regular fa-circle-question text-xs"
            />
          </span>
        </div>
        <div
          class="flex flex-col gap-4"
        >
          <div
            class=""
          >
            <div
              class="input input--select    "
              data-testid="select"
            >
              <label
                class="!translate-y-0 !text-xs"
                for="File source"
              >
                File source
              </label>
              <div
                class=" css-b62m3t-container"
              >
                <span
                  class="css-1f43avz-a11yText-A11yText"
                  id="react-select-2-live-region"
                />
                <span
                  aria-atomic="false"
                  aria-live="polite"
                  aria-relevant="additions text"
                  class="css-1f43avz-a11yText-A11yText"
                />
                <div
                  class="input-select__control css-13cymwt-control"
                >
                  <div
                    class="input-select__value-container input-select__value-container--has-value css-1fdsijx-ValueContainer"
                  >
                    <span
                      class="mr-1 text-sm text-neutral-400"
                    >
                      Git repository
                    </span>
                    <input
                      aria-autocomplete="list"
                      aria-expanded="false"
                      aria-haspopup="true"
                      aria-readonly="true"
                      class="css-mohuvp-dummyInput-DummyInput"
                      id="File source"
                      inputmode="none"
                      role="combobox"
                      tabindex="0"
                      value=""
                    />
                  </div>
                  <div
                    class="input-select__indicators css-1hb7zxy-IndicatorsContainer"
                  >
                    <span
                      class="input-select__indicator-separator css-1u9des2-indicatorSeparator"
                    />
                    <div
                      aria-hidden="true"
                      class="input-select__indicator input-select__dropdown-indicator css-1xc3v61-indicatorContainer"
                    >
                      <svg
                        aria-hidden="true"
                        class="css-tj5bde-Svg"
                        focusable="false"
                        height="20"
                        viewBox="0 0 20 20"
                        width="20"
                      >
                        <path
                          d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
                <input
                  name="File source"
                  type="hidden"
                  value="GIT_REPOSITORY"
                />
              </div>
              <input
                name="File source"
                type="hidden"
                value="GIT_REPOSITORY"
              />
              <div
                class="pointer-events-none absolute right-4 top-1/2 -translate-y-1/2"
              >
                <span
                  class="icon-solid-angle-down shrink-0 text-sm text-neutral-400"
                  role="img"
                />
              </div>
            </div>
          </div>
        </div>
      </form>
    </section>
  </div>
</body>
`;

exports[`ValuesOverrideFilesSetting should match snapshot with YAML type 1`] = `
<body>
  <div>
    <section
      class="flex flex-col"
    >
      <h1
        class="text-neutral-400 text-xl font-bold mb-2"
      >
        Values override as file
      </h1>
      <form
        class="space-y-10"
      >
        <div>
          <p
            class="text-sm text-neutral-350"
          >
            Define the YAML file(s) to be applied as override to the default values.yaml delivered with the chart. It is highly recommended to store the override file(s) in a git repository.
          </p>
          <span
            aria-controls="radix-:rd:"
            aria-expanded="false"
            aria-haspopup="dialog"
            class="cursor-pointer text-sm font-medium text-brand-500 transition hover:text-brand-600"
            data-state="closed"
            type="button"
          >
            How it works 
            <i
              aria-hidden="true"
              class="fa-regular fa-circle-question text-xs"
            />
          </span>
        </div>
        <div
          class="flex flex-col gap-4"
        >
          <div
            class=""
          >
            <div
              class="input input--select    "
              data-testid="select"
            >
              <label
                class="!translate-y-0 !text-xs"
                for="File source"
              >
                File source
              </label>
              <div
                class=" css-b62m3t-container"
              >
                <span
                  class="css-1f43avz-a11yText-A11yText"
                  id="react-select-3-live-region"
                />
                <span
                  aria-atomic="false"
                  aria-live="polite"
                  aria-relevant="additions text"
                  class="css-1f43avz-a11yText-A11yText"
                />
                <div
                  class="input-select__control css-13cymwt-control"
                >
                  <div
                    class="input-select__value-container input-select__value-container--has-value css-1fdsijx-ValueContainer"
                  >
                    <span
                      class="mr-1 text-sm text-neutral-400"
                    >
                      Git repository
                    </span>
                    <input
                      aria-autocomplete="list"
                      aria-expanded="false"
                      aria-haspopup="true"
                      aria-readonly="true"
                      class="css-mohuvp-dummyInput-DummyInput"
                      id="File source"
                      inputmode="none"
                      role="combobox"
                      tabindex="0"
                      value=""
                    />
                  </div>
                  <div
                    class="input-select__indicators css-1hb7zxy-IndicatorsContainer"
                  >
                    <span
                      class="input-select__indicator-separator css-1u9des2-indicatorSeparator"
                    />
                    <div
                      aria-hidden="true"
                      class="input-select__indicator input-select__dropdown-indicator css-1xc3v61-indicatorContainer"
                    >
                      <svg
                        aria-hidden="true"
                        class="css-tj5bde-Svg"
                        focusable="false"
                        height="20"
                        viewBox="0 0 20 20"
                        width="20"
                      >
                        <path
                          d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
                <input
                  name="File source"
                  type="hidden"
                  value="GIT_REPOSITORY"
                />
              </div>
              <input
                name="File source"
                type="hidden"
                value="GIT_REPOSITORY"
              />
              <div
                class="pointer-events-none absolute right-4 top-1/2 -translate-y-1/2"
              >
                <span
                  class="icon-solid-angle-down shrink-0 text-sm text-neutral-400"
                  role="img"
                />
              </div>
            </div>
          </div>
        </div>
        <section
          class="flex flex-col"
        >
          <h2
            class="text-neutral-400 text-base font-bold mb-2"
          >
            Override with raw Yaml
          </h2>
          <p
            class="mb-6 text-sm text-neutral-350"
          >
            You can define here the YAML containing the overrides you want to apply. The YAML will be stored by Qovery and can be updated later within the settings but no history will be retained.
          </p>
          <div
            class="flex flex-col gap-3"
          >
            <div
              class="mb-5 rounded border border-neutral-250 bg-neutral-100 w-full"
              data-testid="block-content"
            >
              <div
                class="flex h-9 items-center justify-between border-b border-neutral-250 px-4"
              >
                <h2
                  class="text-ssm font-medium text-neutral-400"
                >
                  Raw YAML
                </h2>
              </div>
              <div
                class="p-0"
              >
                <div
                  class="my-4 px-10 py-5 text-center"
                >
                  <i
                    aria-hidden="true"
                    class="fa-solid fa-wave-pulse text-neutral-350"
                  />
                  <p
                    class="mb-3 mt-1 text-xs font-medium text-neutral-350"
                  >
                    No override defined.
                  </p>
                  <button
                    class="inline-flex items-center shrink-0 font-medium transition focus-visible:[&:not(:active)]:outline-2 outline-0 shadow-sm disabled:shadow-none active:shadow-none outline-neutral-500 text-sm h-9 px-3 rounded bg-brand-500 active:bg-brand-600 hover:bg-brand-400 text-white disabled:text-brand-300 disabled:bg-brand-100"
                    type="button"
                  >
                    Create override 
                    <i
                      aria-hidden="true"
                      class="fa-solid fa-pen ml-2"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </section>
      </form>
    </section>
  </div>
</body>
`;
